### PR TITLE
[Enhancement] Optimize primary replica selection

### PR DESCRIPTION
**Pull request by meegoo**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

At present, we use a `random algorithm` to select the primary replicas in the replicated storage to distribute them as evenly as possible on each BE. Now we use a simple `greedy algorithm` to improve the distribution result.
## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 10 / 10 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|10|10|100.00%|[]|

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/planner/OlapTableSink.java [hunk 1]
```diff
@@ -345,9 +347,22 @@ private TOlapTableLocationParam createLocation(OlapTable table) throws UserExcep
                                             + tablet.getId() + ", backends: " +
                                             Joiner.on(",").join(localTablet.getBackends()));
                         }
-                        // replicas[0] will be the primary replica
+
                         List<Long> replicas = Lists.newArrayList(bePathsMap.keySet());
-                        Collections.shuffle(replicas);
+                        int lowUsageIndex = 0;
```
Comment (rickif) [inline fe/fe-core/src/main/java/com/starrocks/planner/OlapTableSink.java:352]: Is this to pick out the BE with the least primaries? Maybe it's better to add some comments to make it clear.

Comment (meegoo) [reply fe/fe-core/src/main/java/com/starrocks/planner/OlapTableSink.java:352]: There will be a PR to implement the backlist strategy soon, comment will be made at that time.

**Review by chaoyli — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Pull request merged**